rfc/20240513-static-evaluation-providers.md

```diff
@@ -329,22 +493,29 @@ terraform {
     }
 }
 ```
+
 Initial discussion with Core Team suggests that this should be moved to it's own issue as it would be useful, but does not impact the work done in this issue.
 
+#### Dynamic or early-evaluated provider aliases
 
-There's also an ongoing discussion on allowing variable names in provider aliases.
+Early discussion for this feature considered possibly allowing more than just constant strings in the `alias` argument. If we supported this it would likely be limited only to early-evaluation to preserve the requirement that each `provider` block have a known alias when we build the main language runtime's dependency graph.
 
 Example:
 ```hcl
-# Why would you want to do this?  It looks like terraform deprecated this some time after 0.11.
+# Why would you want to do this?
 provider "aws" {
   alias = var.foo
 }
 ```
-Are there any downsides to supporting this?
+
+We have intentionally omitted this for now because we don't know of any real use-cases for it, and it's conceptually easier to imagine `alias` as being effectively equivalent to the second label in the header of a `resource` block -- a statically-chosen unique identifier for use in references elsewhere -- than as another possibly-dynamic element in addition to instance keys.
+
+We can safely add early-eval-based aliases in a later release without any breaking changes, so we will await specific use-cases for this before we decide whether and how to support it.
 
 ### Future Considerations
 
+A potential fully-dynamic version of this feature is discussed in another RFC: [Dynamic Provider Instances and Instance Assignment](./20241021-dynamic-provider-instances.md). We don't intend to support that immediately but have "designed ahead" to reduce the risk that backward-compatibility with the static eval implementation would block a later dynamic implementation.
```

Comment:
Link to https://github.com/opentofu/opentofu/pull/2088 for now and update the link later.